src/service_providers/index.js

```diff
@@ -1,31 +1,29 @@
-import fs from 'fs';
+import fsApi from 'fs';
+const fs = fsApi.promises;
 import path from 'path';
 
-import dotenv from 'dotenv';
-dotenv.config();
-
-const __dirname = path.dirname(new URL(import.meta.url).pathname);
-const dirPath = path.resolve(__dirname, '../..', process.env.NODE_ENV === 'test' ? 'test' : '', 'providers');
-const sanitizers = {};
-
-export default async function getServiceProviders() {
+export default async function loadServiceProviders(dirPath) {
   const serviceProviders = {};
 
-  const filenames = fs.readdirSync(dirPath);
+  const filenames = await fs.readdir(dirPath);
 
   for (let filename of filenames) {
     if (filename.indexOf('.sanitizers.js') !== -1) {
       const serviceProviderId = path.basename(filename, '.sanitizers.js');
-      sanitizers[serviceProviderId] = await import(path.join(dirPath, filename));
+
+      serviceProviders[serviceProviderId] = {
+        ...serviceProviders[serviceProviderId],
+        sanitizers: await import(path.join(dirPath, filename)),
+      };
     } else if (filename.indexOf('.json') !== -1) {
       const serviceProviderId = path.basename(filename, '.json');
-      serviceProviders[serviceProviderId] = JSON.parse(fs.readFileSync(path.join(dirPath, filename)));
+
+      serviceProviders[serviceProviderId] = {
+        ...serviceProviders[serviceProviderId],
+        ...JSON.parse(await fs.readFile(path.join(dirPath, filename))),
+      };
```

Comment:
> It would probably be easier to understand if we had a first loop that loaded all the JSON definitions, and then another one that only added the sanitizers 🙂
> Checking the _extension_ rather than the _presence_ (`indexOf !== -1`) would also be more robust.